design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,147 @@
+treat this user spec as a declarative intent for the cluster so long as the
+`removeFromSpecAfterCompletion` is unspecified or set to `false`. Setting this value to `true` will
+change Rook's behavior to treat the spec imperatively, intended for use by the Orchestrator Mgr
+Module. Upon completion of the `ceph-volume` command with the Drive Group spec, Rook will remove the
+Drive Group item from the `driveGroups` spec for the corresponding node in the `CephCluster`
+resource so that Rook will not rerun the creation again, which would break the Ceph Orchestrator Mgr
+Module's imperative intention. The Mgr Module should obviously set this value to `true` for all
+operations.
+
+**TODO DISCUSS THIS:**
+The Drive Group item should be removed regardless of whether the `ceph-volume` command result was
+successful or unsuccessful. Rook should report the success/fail result back to the Rook
+Orchestrator Mgr Module via a `CephCluster` status mechanism so that the Orchestrator can take
+appropriate actions to notify the user of failure. The status should be cleared by the Rook
+Orchestrator Mgr Module when the user dismisses the status message, and Rook should never
+clear statuses on its own, only appending new statuses to a list.
+
+```yaml
+# first draft suggestions for status mechanism
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+status: # preexisting
+  driveGroupOSDCreations: # proposed new
+    - node: <node name> # proposed new
+      driveGroupSpec: < JSON string to pass to ceph-volume for OSD creation >
+      success: < bool >
+      message: < string message should be filled in if success==false >
+```
+
+### Removing OSDs
+When removing an OSD or OSDs via the Rook Orchestrator Module, the Ceph mgr running in Kubernetes
+will will also perform whatever Ceph operations are necessary to remove the OSD as would be expected
+by the user. This will replace the manual, imperative steps currently required by Rook-Ceph users to
+remove OSDs from the storage cluster documented
+[here for Rook v1.2](https://rook.io/docs/rook/v1.2/ceph-osd-mgmt.html#remove-an-osd).
+
+There is currently a proposal for a shared Ceph Mgr Orchestrator Module implementing common OSD
+removal steps [[link]](https://github.com/ceph/ceph/pull/32677) usable by any other Ceph Mgr
+Orchestrator module. The OSD removal mechanism should use this where possible.
+
+Rook will not automatically remove Kubernetes Deployments for OSDs unless
+`removeOSDsIfOutAndSafeToRemove: true`. This should be documented for users, and users should be
+advised to set this value to `true`.
+
+The `CephCluster` CRD allows users to add disks to nodes in several places including disks on all
+nodes in the Kubernetes cluster, to a specific node, or via disk pattern matching (in two different
+ways). To avoid the complexity of having to manage the logic for determining how to remove a node
+from the\ `CephCluster` resource, OSD removal will **not** alter the `CephCluster` resource. While
```

Comment:
- Agreed, documentation will have to cover this case. Users should be aware of how to remove devices or nodes so the settings won't re-add the OSDs.
- typo: `the\`